File: kickr-client/src/services/reportService.ts
import axiosInstance from './axios';

export type ReportType = 'MATCH_REVIEW' | 'COMMENT';
export type ReportReason = 'SPAM' | 'INAPPROPRIATE_LANGUAGE' | 'HARASSMENT' | 'OTHER';
export type ReportStatus = 'PENDING' | 'RESOLVED' | 'REJECTED';

export interface CreateReportDto {
    targetType: ReportType;
    targetId: string;
    reason: ReportReason;
    description?: string;
}

export interface Report {
    id: string;
    reporter: any;
    targetType: ReportType;
    targetId: string;
    reason: ReportReason;
    description: string;
    status: ReportStatus;
    createdAt: string;
}

export const reportService = {
    createReport: async (dto: CreateReportDto) => {
        const response = await axiosInstance.post<Report>('/reports', dto);
        return response.data;
    },

    getAllReports: async (status?: ReportStatus) => {
        const response = await axiosInstance.get<Report[]>('/reports', {
            params: { status }
        });
        return response.data;
    },

    updateStatus: async (reportId: string, status: ReportStatus) => {
        const response = await axiosInstance.put<Report>(`/reports/${reportId}/status`, null, {
            params: { status }
        });
        return response.data;
    },

    deleteReport: async (reportId: string) => {
        await axiosInstance.delete(`/reports/${reportId}`);
    }
};

File: kickr-server/src/main/java/com/kickr_server/admin/report/ReportService.java
package com.kickr_server.admin.report;

import com.kickr_server.dto.admin.CreateReportDto;
import com.kickr_server.dto.admin.ReportDto;
import com.kickr_server.user.User;
import com.kickr_server.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class ReportService {

    private final ReportRepository reportRepository;
    private final UserRepository userRepository;

    @Transactional
    public ReportDto createReport(String email, CreateReportDto dto) {
        User reporter = userRepository.findByName(email)
                .or(() -> userRepository.findByEmail(email))
                .orElseThrow(() -> new RuntimeException("Reporter not found"));

        Report report = Report.builder()
                .reporter(reporter)
                .targetType(dto.getTargetType())
                .targetId(dto.getTargetId())
                .reason(dto.getReason())
                .description(dto.getDescription())
                .status(ReportStatus.PENDING)
                .build();

        return ReportDto.fromEntity(reportRepository.save(report));
    }

    public List<ReportDto> getAllReports() {
        return reportRepository.findAllByOrderByCreatedAtDesc().stream()
                .map(ReportDto::fromEntity)
                .collect(Collectors.toList());
    }

    public List<ReportDto> getReportsByStatus(ReportStatus status) {
        return reportRepository.findByStatusOrderByCreatedAtDesc(status).stream()
                .map(ReportDto::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional
    public ReportDto updateStatus(UUID reportId, ReportStatus status) {
        Report report = reportRepository.findById(reportId)
                .orElseThrow(() -> new RuntimeException("Report not found"));

        report.setStatus(status);
        return ReportDto.fromEntity(reportRepository.save(report));
    }

    @Transactional
    public void deleteReport(UUID reportId) {
        if (!reportRepository.existsById(reportId)) {
            throw new RuntimeException("Report not found");
        }
        reportRepository.deleteById(reportId);
    }
}
